packages/tests/src.ts/test-wallet.spec.ts

```diff
@@ -605,4 +603,51 @@ describe("Wallet createAccount", function () {
         assert.strictEqual(acc2BalanceBefore < acc2BalanceAfter, true);
         assert.strictEqual(acc2BalanceAfter - acc2BalanceBefore, 100000000);
     }).timeout(timeout);
+
+    it("Should throw an error for crypto transfer with data field", async function() {
```

Comment:
Can we add e2es that catch the other scenarios as-well:
- `isCryptoTransfer = false` (when specifying only `to` and `value`) thus it makes a `ContractCall` 
- `isCryptoTransfer = undefined` (when specifying only `to` and `value`) and if `to` is contract, it does `ContractCall`
- `isCryptoTransfer = undefined` (when specifying only `to` and `value`) and if `to` is account, it does `CryptoTransfer`
- `isCryptoTransfer = undefined` (when specifying only `to` and `value`) and if provider is not set, it throws an error